### DISCUSSION TITLE
Move data from standalone Fuseki server to Fuseki HA setup

### DISCUSSION THREAD

**Discussion by sempike** [afs/rdf-delta#129] [Q&A] [answered]
Hi,

I have a Fuseki HA setup deployed to an AKS cluster and I am about to do some performance tests.
There is already a Standalone Fuseki server with quite much of test data. Is there a simple way to import that data to Fuseki HA?

I already tried to create a backup in the standalone Fuseki server and then to import it in one of the HA Fuseki server replicas with _tdbloader_ , but it did not work. 
I am wondering if it can work and I’m messing up missing something or not.

**Comment by afs** [comment, marked as answer]
Hi @sempike,

Yes - with care! There is no need to start the databases in the Fuseki servers from empty. The starting state can be a database copy (blank nodes must be the same).

To do that, build the database on one machine, shutdown down Fuseki or any TDB code using it.  Now the database directory can be copied to the Fuseki HS setup (server not running, no patches) and overwrite or place initially the database on each Fuseki HA server.

Example assembler file for externally placed dataset -- property `delta:dataset` points to any dataset description.

[rdf-delta-integration-tests/testing/assembler/delta-assembler-ext-good-1.ttl](https://github.com/afs/rdf-delta/blob/main/rdf-delta-integration-tests/testing/assembler/delta-assembler-ext-good-1.ttl).

If you load separate databases per server, the databases and have blank nodes, the internal system ids of the blank nodes won't be the same.

With care, the same approach - and copying the Zone state - can be used to add a Fuseki HA server but the cluster must be shutdown at the time.
